libcnb-data/src/launch.rs

```diff
@@ -65,30 +66,102 @@ pub struct Label {
 pub struct Process {
     pub r#type: ProcessType,
     pub command: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub args: Option<Vec<String>>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub direct: Option<bool>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub default: Option<bool>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub args: Vec<String>,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub direct: bool,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub default: bool,
+}
+
+pub struct ProcessBuilder {
+    process: Process,
 }
 
-impl Process {
-    pub fn new(
-        r#type: ProcessType,
-        command: impl Into<String>,
-        args: Option<impl IntoIterator<Item = impl Into<String>>>,
-        direct: Option<bool>,
-        default: Option<bool>,
-    ) -> Self {
+/// A non-consuming builder for [`Process`] values.
+///
+/// # Examples
+/// ```
+/// # use libcnb_data::process_type;
+/// # use libcnb_data::launch::ProcessBuilder;
+/// ProcessBuilder::new(process_type!("web"), "java")
+///     .arg("-jar")
+///     .arg("target/application-1.0.0.jar")
+///     .default(true)
+///     .build();
+/// ```
+impl ProcessBuilder {
+    /// Constructs a new `ProcessBuilder` with the following defaults:
+    ///
+    /// * No arguments to the process
+    /// * `direct` is `false`
+    /// * `default` is `false`
+    pub fn new(r#type: ProcessType, command: impl Into<String>) -> Self {
         Self {
-            r#type,
-            command: command.into(),
-            args: args.map(|args| args.into_iter().map(std::convert::Into::into).collect()),
-            direct,
-            default,
+            process: Process {
+                r#type,
+                command: command.into(),
+                args: Vec::new(),
+                direct: false,
+                default: false,
+            },
         }
     }
+
+    /// Adds an argument to the process.
+    ///
+    /// Only one argument can be passed per use. So instead of:
```

Comment:
(nit)
"can be passed per use" suggests there is validation. libcnb doesn't validate this, so I presume it's more of a "should"? (Unless Pack/lifecycle do validate?)

```suggestion
    /// Only one argument should be passed per use. So instead of:
```

Reply:
This is a verbatim copy of the `Command` function from the standard lib with the same name. I liked that it 100% fits what we're doing here. If you don't feel strongly, I'd like to keep it the way it is.

Reply:
Ah, then works for me :-)